build.js

```diff
@@ -9,26 +10,36 @@ const stringifyObject = require('stringify-object');
 
 const pkg = require('./package.json');
 
-let array = Object.keys(require('spdx-license-list'));
-let files = [
-  {
-    path: pkg.main,
-    contents: JSON.stringify(array, null, '  ')
-  },
-  {
-    path: require('./bower.json').main,
-    contents: `window.spdxLicenseIds = ${stringifyObject(array, {indent: '  '})};`
-  }
-];
+http.get('http://spdx.org/licenses/licenses.json', response => {
+  let buffer = '';
+  response
+    .on('data', data => {
+      buffer += data;
+    })
+    .on('end', () => {
+      const json = JSON.parse(buffer);
+      let array = json.licenses.map(license => license.licenseId).sort();
```

Comment:
I don't believe the order of the identifiers has any significance.

If anything, I hoped sorting would keep the diff short if and when you run the script to update the JSON files.



On Thu, May 21, 2015 at 3:54 AM -0700, "Shinnosuke Watanabe" <notifications@github.com<mailto:notifications@github.com>> wrote:


In build.js<https://github.com/shinnn/spdx-license-ids/pull/2#discussion_r30791404>:

> -    contents: JSON.stringify(array, null, '  ')
> -  },
> -  {
> -    path: require('./bower.json').main,
> -    contents: `window.spdxLicenseIds = ${stringifyObject(array, {indent: '  '})};`
> -  }
> -];
> +http.get('http://spdx.org/licenses/licenses.json', response => {
> +  let buffer = '';
> +  response
> +    .on('data', data => {
> +      buffer += data;
> +    })
> +    .on('end', () => {
> +      const json = JSON.parse(buffer);
> +      let array = json.licenses.map(license => license.licenseId).sort();


Do you think this .sort() is needed? In the current version I respect the original order.

-
Reply to this email directly or view it on GitHub<https://github.com/shinnn/spdx-license-ids/pull/2/files#r30791404>.